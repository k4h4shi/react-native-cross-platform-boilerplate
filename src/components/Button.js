// @flow
import React, { Component } from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";

type Props = {
  title: string,
  onPress: () => {}
};

export default ({ title, onPress }: Props) => (
  <TouchableOpacity onPress={onPress} style={styles.button}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center"
  },
  title: {
    backgroundColor: "transparent",
    borderStyle: "solid",
    borderColor: "black",
    borderRadius: 3,
    borderWidth: 1.5,
    padding: 6,
    color: "black"
  }
});
