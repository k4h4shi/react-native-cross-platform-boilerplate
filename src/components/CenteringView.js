// @flow
import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

type Props = { children: {} };

export default ({ children }: Props) => (
  <View style={styles.centering}>{children}</View>
);

const styles = StyleSheet.create({
  centering: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  }
});
