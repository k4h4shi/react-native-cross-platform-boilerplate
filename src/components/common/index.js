export { default as Button } from "./Button";
export { default as CenteringView } from "./CenteringView";
