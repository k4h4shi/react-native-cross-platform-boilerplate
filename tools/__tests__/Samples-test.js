// __tests__/App-test.js
import React from "react";
import data from "../data";
import { Samples } from "../../src/components/samples";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Samples samples={data.samples} />).toJSON();
  expect(tree).toMatchSnapshot();
});
