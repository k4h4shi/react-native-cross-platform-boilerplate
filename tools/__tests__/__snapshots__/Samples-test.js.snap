// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1.5,
        "margin": 5,
        "padding": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      id: 
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      title: 
      title1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      description: 
      description1
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1.5,
        "margin": 5,
        "padding": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      id: 
      2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      title: 
      title2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      description: 
      description2
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1.5,
        "margin": 5,
        "padding": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      id: 
      3
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      title: 
      title3
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      description: 
      description3
    </Text>
  </View>,
]
`;
